### PR TITLE
[VSphere] Tsdb enabled release

### PR DESCRIPTION
**Pull request by ritalwar**
- Enhancement

## What does this PR do?

This PR enables TSDB for Vsphere.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Relates #6152

### CODE REVIEW

**Comment by lalit-satapathy** [comment]
@ritalwar, There are changes related to sample event. Could you please summarise, whats triggering it?

**Comment by ritalwar** [comment]
> @ritalwar, There are changes related to sample event. Could you please summarise, whats triggering it?

I ran tests on this package with --generate command, which updated the sample_events.

**Comment by lalit-satapathy** [comment]
> > @ritalwar, There are changes related to sample event. Could you please summarise, whats triggering it?
> 
> I ran tests on this package with --generate command, which updated the sample_events.

@ishleenk17 can you review the sample event changes, is this expected?

File: packages/vsphere/data_stream/virtualmachine/sample_event.json [hunk 1]
```diff
@@ -1,42 +1,39 @@
...
+        "version": "8.9.0"
     },
     "data_stream": {
         "dataset": "vsphere.virtualmachine",
         "namespace": "ep",
         "type": "metrics"
     },
     "ecs": {
-        "version": "8.5.1"
+        "version": "8.0.0"
```
Comment (ishleenk17) [inline packages/vsphere/data_stream/virtualmachine/sample_event.json:16]: Why has this ecs version changed?

**Comment by ishleenk17** [comment]
> > > @ritalwar, There are changes related to sample event. Could you please summarise, whats triggering it?
> > 
> > 
> > I ran tests on this package with --generate command, which updated the sample_events.
> 
> @ishleenk17 can you review the sample event changes, is this expected?

Yes these are expected. Just a question regarding ecs version that I have wanted to clarify.

File: packages/vsphere/data_stream/virtualmachine/sample_event.json (hunk 1 above)
Comment (ritalwar) [reply packages/vsphere/data_stream/virtualmachine/sample_event.json]: The version was upgraded using script as part of updating ecs version. But when sample_event.json was generated using elastic-package in this PR it did not take version specified in build.yml into account.
I have observed in other integrations, pipeline is used to set ecs version in documents, so I have now added pipeline to update the version and even upgraded it to the latest one.

**Comment by lalit-satapathy** [comment]
> > > > @ritalwar, There are changes related to sample event. Could you please summarise, whats triggering it?
> > > 
> > > 
> > > I ran tests on this package with --generate command, which updated the sample_events.
> > 
> > 
> > @ishleenk17 can you review the sample event changes, is this expected?
> 
> Yes these are expected. Just a question regarding ecs version that I wanted to clarify.

Is this thread closed now?

File: packages/vsphere/data_stream/virtualmachine/sample_event.json (hunk 1 above)
Comment (ishleenk17) [reply packages/vsphere/data_stream/virtualmachine/sample_event.json]: In short are we saying that ecs version is picked up from pipeline and not from build.yml ?

Comment (ritalwar) [reply packages/vsphere/data_stream/virtualmachine/sample_event.json]: The version of ECS from build.yml is mainly used to import definitions.
The ecs version in the document is probably coming from beats, so if we want to set some specific version in documents it is done through pipeline.

Comment (ishleenk17) [reply packages/vsphere/data_stream/virtualmachine/sample_event.json]: Something doesn't add up.
If it's the version of the beats, why did it pick 8.5.1 (the older build.yml ecs version) before this change? It should have picked 8.0.0 then as well.

Comment (ritalwar) [reply packages/vsphere/data_stream/virtualmachine/sample_event.json]: The ecs version for VSphere and most packages was updated as part of this: https://github.com/elastic/integrations/pull/5052.
 
This wasn't done using elastic-package test generate command like in this PR.
And also if you notice the version 8.5.1 -> 8.0.0 in this PR got changed only for datastore and vm datastream as for other two there already existed a pipeline which was setting ecs version to 8.5.1 and it remained unchanged.

Also refer: https://elastic.slack.com/archives/C0349GN4S2H/p1688050743542259?thread_ts=1688050595.028039&cid=C0349GN4S2H

**Comment by ishleenk17** [comment]
> > > > > @ritalwar, There are changes related to sample event. Could you please summarise, whats triggering it?
> > > > 
> > > > 
> > > > I ran tests on this package with --generate command, which updated the sample_events.
> > > 
> > > 
> > > @ishleenk17 can you review the sample event changes, is this expected?
> > 
> > 
> > Yes these are expected. Just a question regarding ecs version that I wanted to clarify.
> 
> Is this thread closed now?

Yes, we can close this thread now.

**Review by ishleenk17 — APPROVED** [review]
Approved changes from non-TSDB perspective.

**Review by lalit-satapathy — APPROVED** [review]

**Pull request merged**